app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingService.java

```diff
@@ -546,6 +549,14 @@ private void shutdown(State newState, SyncthingRunnable.OnSyncthingKilled onKill
             }
             mSyncthingRunnable = null;
         }
+        if (startupTaskIsRunning()) {
+            mStartupTask.cancel(true);
+            Log.v(TAG, "Waiting for mStartupTask to finish after cancelling ...");
+            try {
+                mStartupTask.get();
+            } catch (Exception e) { }
+            mStartupTask = null;
+        }
```

Comment:
I assume this is a bug fix for rapid stop/start leading to some bad state?

Reply:
Yes. I got into such a state frequently with the db reset on startup, which causes a restart.